### DISCUSSION TITLE
online and offline store clarification

### DISCUSSION THREAD

**Discussion by theprofessorvijay** [feast-dev/feast#1776] [Q&A] [answered]
Hello,

I understand that core service is used for registering the metadata about features, I am not clear about what is the role of serving in Feast, 
for example I have created feast client as below, my core is connected to **PostgreSQL** and serving is connected to **Redis**

```python
client = Client(core_url='localhost:6565', serving_url='localhost:6566', project="demo")
```

While getting **FeatureStore** why is that we need to specify online and offline store configuration details like redis connection string, is not supposed to connect to **Redis** thorough serving URL for loading the data for online serving? similarly we have provided batch_source while creating **FeatureTable**, is not enough to get the data for offline training (historical data)?

In a production environment, sharing **Redis** credentials not an option to create **FeatureStore**

Please clarify

Note: I have gone through the documentation, still do not have clarity

**Comment by woop** [comment, marked as answer]
@theprofessorvijay Can you please use the latest version of Feast? Feast 0.9 is not supported any more.
